src/App.tsx

```diff
@@ -1,8 +1,88 @@
-import React from 'react';
-import './App.css';
+import React, { FC, useState } from 'react';
+import { TodoList } from './TodoList/TodoList';
+import { TodoProps, UserProps, UsersTodo } from './types';
+import { getUsers, getTodos } from './api/api';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: FC = () => {
+  const [todos, setTodos] = useState<UsersTodo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<UsersTodo[]>([...todos]);
+  const [isLoading, setLoading] = useState(false);
+
+  const loadTodos = async () => {
+    setLoading(true);
+
+    const [todosFromServer, users] = await Promise.all([getTodos(), getUsers()]);
+
+    const preparedTodos = todosFromServer.map((todo: TodoProps) => {
+      const user = users.find((person: UserProps) => person.id === todo.userId) as UserProps;
+
+      return {
+        ...todo,
+        user,
+      };
+    });
+
+    setTodos(preparedTodos);
+    setLoading(false);
+    setFilteredTodos(preparedTodos);
+  };
+
+  const filterTodo = (typeOfFilter: string) => {
+    switch (typeOfFilter) {
+      case 'byTitle':
+        setFilteredTodos([...todos]
+          .sort((a, b) => a.title.localeCompare(b.title)));
+        break;
+      case 'byName':
+        setFilteredTodos([...todos]
+          .sort((a, b) => a.user.name.localeCompare(b.user.name)));
+        break;
+      case 'byCompleted':
+        setFilteredTodos([...todos]
+          .sort((a, b) => b.completed.toString()
+            .localeCompare(a.completed.toString())));
+        break;
```

Comment:
не читабельно на 100% 
```suggestion
      case 'byCompleted': {
        const sortedTodos = [...todos].sort((a, b) => (
          b.completed.toString().localeCompare(a.completed.toString())
        ));
        
        setSortedTodos[sortedTodos];
        break;
      }
```